lib/queue_classic_plus/worker.rb

```diff
@@ -16,23 +16,17 @@ def handle_failure(job, e)
       QueueClassicPlus.logger.info "Handling exception #{e.message} for job #{job[:id]}"
 
       force_retry = false
-      begin
+      if e.respond_to?(:original_exception) && e.original_exception.is_a?(PG::Error)
         # If we've got here, unfortunately ActiveRecord's rollback mechanism may
         # not have kicked in yet and we might be in a failed transaction. To be
         # *absolutely* sure the retry/failure gets enqueued, we do a rollback
         # just in case (and if we're not in a transaction it will be a no-op).
-        QC.default_conn_adapter.execute 'ROLLBACK'
-      rescue PG::UnableToSend, PG::ConnectionBad => e
+        QueueClassicPlus.logger.info "Reset connection for job #{job[:id]}"
+        @conn_adapter.connection.reset
```

Comment:
In my understanding, `@conn_adapter.connection` is initialized with `ActiveRecord::Base.connection.raw_connection`, so `reset` is actually trying to re-connect `ActiveRecord::Base.connection.raw_connection`. I'll try to make sure this.